app/src/main/java/ch/epfl/sweng/eventmanager/repository/impl/FirebaseHelper.java

```diff
@@ -88,4 +88,13 @@ public static UploadTask uploadFileToStorage(StorageReference ref, Uri fileUri,
 
         return dbRef.setValue(elem);
     }
+
+    public static <T> Task<Void> replaceElement(int eventId, String ref, T elem){
+        DatabaseReference dbRef = FirebaseDatabase.getInstance()
+                .getReference(ref)
+                .child("event_" + eventId)
+                .push(); // Create a new key in the list
+
+        return dbRef.setValue(null).addOnCompleteListener(command -> dbRef.setValue(elem));
```

Comment:
the first part is useless: you only need to do dbRef.setValue(elem)

Reply:
Okay ! I will change that !